include/swift/AST/AccessScope.h

```diff
@@ -103,7 +103,7 @@ class AccessScope {
     return false;
   }
 
-  bool isInternalOrLess() const { return getDeclContext() != nullptr; }
+  bool hasScope() const { return getDeclContext() != nullptr; }
```

Comment:
nit: `isInternalOrLess()` seems to be a better name for what this implies.

Reply:
If necessary, we could define an `isPackageOrMore()` function that negates this.

Reply:
Turns out this doesn't actually check the restrictiveness of the access level between two symbols. I've added a doc comment with a code example to illustrate the meaning of this function. I've also renamed it as `isInContext`. Lmk if it makes it clearer or otherwise.

Reply:
Sounds reasonable to me. Thank you, Ellie!